### PR TITLE
CB-16465 Lazy load safe toString() methods

### PR DESCRIPTION
**Pull request by Bajzathd**
Added tests for Entity classes to make sure they implement toString() correctly.

### CODE REVIEW

File: core-model/src/main/java/com/sequenceiq/cloudbreak/domain/StructuredEventEntity.java [hunk 1]
```diff
@@ -133,4 +133,17 @@ public String getResourceCrn() {
...
+    @Override
+    public String toString() {
+        return "StructuredEventEntity{" +
+                "id=" + id +
+                ", eventType=" + eventType +
+                ", resourceType='" + resourceType + '\'' +
+                ", resourceId=" + resourceId +
+                ", resourceCrn='" + resourceCrn + '\'' +
+                ", timestamp=" + timestamp +
+                ", structuredEventJson=" + structuredEventJson +
```
Comment (doktoric) [inline core-model/src/main/java/com/sequenceiq/cloudbreak/domain/StructuredEventEntity.java:146]: do we need to the whole json ?

**Comment by doktoric** [comment]
it would be nice to see unit tests just to make sure we dont break this

**Comment by Bajzathd** [comment]
> it would be nice to see unit tests just to make sure we dont break this

I have added EntityToStringTest for this purpose, do we need other tests?

**Comment by Bajzathd** [comment]
I have realized that my initial test only covered the entities included in the core service, so I have moved the original unit test to the common module, and extended it in each of our services (`FreeipaEntityToStringTest`, `FreeipaEntityToStringTest`, etc)

File: core-model/src/main/java/com/sequenceiq/cloudbreak/domain/SecurityConfig.java [hunk 2]
```diff
@@ -144,4 +145,19 @@ public String getUserFacingKey() {
     public void setUserFacingKey(String userFacingKey) {
         this.userFacingKey = new Secret(userFacingKey);
     }
+
+    @Override
+    public String toString() {
+        return "SecurityConfig{" +
+                "id=" + id +
+                ", clientKey=" + clientKey +
```
Comment (lnardai) [inline core-model/src/main/java/com/sequenceiq/cloudbreak/domain/SecurityConfig.java:153]: Do we want to log these? these are either sensitive info or vault paths, right?
clientKey
userFacingCert
userFacingKey

**Comment by lnardai** [comment]
I'm a little concerned about the extended pressure on the logging system. Might worth to check how much more we log during mock tests.

File: core-model/src/main/java/com/sequenceiq/cloudbreak/domain/SecurityConfig.java (hunk 2 above)
Comment (Bajzathd) [reply core-model/src/main/java/com/sequenceiq/cloudbreak/domain/SecurityConfig.java:153]: Nice catch, I will remove them

**Comment by Bajzathd** [comment]
I do not think that it will have a huge impact, and it is better to log at least the crns of domain objects. But I will compare the IT log size to be sure.

**Comment by Bajzathd** [comment]
@lnardai IT log sizes:
[integcb-logs.tar.gz]	175.02 MB	 
[test.out] 69.18 MB	
These are pretty similar to previous runs' file sizes

**Review by lajosrodek — CHANGES_REQUESTED** [review]
Nice work! A few minor comments.

**Comment by lajosrodek** [comment]
@Bajzathd  A general note: I suggest creating a wiki and sending a msg to Slack regarding the usage patterns & expectations introduced here.

File: environment/src/main/java/com/sequenceiq/environment/network/dao/domain/BaseNetwork.java [hunk 3]
```diff
@@ -265,7 +267,7 @@ public String toString() {
         return "BaseNetwork{" +
                 "id=" + id +
                 ", name='" + name + '\'' +
-                ", environment=" + environment.getName() +
+                ", environment=" + getIfNotNull(environment, EnvironmentView::getName) +
```
[the remark this answers is not in the document]
Comment (Bajzathd) [reply environment/src/main/java/com/sequenceiq/environment/network/dao/domain/BaseNetwork.java:270]: It is not lazy loaded, I only added the getIfNotNull() just to be extra safe

**Comment by Bajzathd** [comment]
Thank you @lajosrodek for the review! I have answered a question and fixed the others. The wiki is a good idea, I will create it shortly.

**Comment by Bajzathd** [comment]
Wiki page is created: https://cloudera.atlassian.net/wiki/spaces/ENG/pages/9980936961/Guideline+for+entity+toString+implementation

**Comment by lnardai** [comment]
Should we close this or you plan to resolve conflicts? Should we this a draft in meanwhile?

**Comment by Bajzathd** [comment]
I'm closing this

**Pull request closed without merging**